internal/provider/kubernetes/controller.go

```diff
@@ -116,6 +116,16 @@ func newGatewayAPIController(mgr manager.Manager, cfg *config.Server, su Updater
 type resourceMappings struct {
 	// Map for storing namespaces for Route, Service and Gateway objects.
 	allAssociatedNamespaces map[string]struct{}
+	// Map for storing TLSRoutes' NamespacedNames attaching to various Gateway objects.
+	allAssociatedTLSRoutes map[string]struct{}
+	// Map for storing HTTPRoutes' NamespacedNames attaching to various Gateway objects.
+	allAssociatedHTTPRoutes map[string]struct{}
+	// Map for storing GRPCRoutes' NamespacedNames attaching to various Gateway objects.
+	allAssociatedGRPCRoutes map[string]struct{}
+	// Map for storing TCPRoutes' NamespacedNames attaching to various Gateway objects.
+	allAssociatedTCPRoutes map[string]struct{}
+	// Map for storing UDPRoutes' NamespacedNames attaching to various Gateway objects.
+	allAssociatedUDPRoutes map[string]struct{}
```

Comment:
replace this with sets.set[string]

Reply:
tracked with https://github.com/envoyproxy/gateway/issues/3411